web/app/scripts/details/details-reference-controller.js

```diff
@@ -1,33 +1,25 @@
 (function () {
     'use strict';
 
-    /* ngInject */
-    function DetailsReferenceController($translate) {
+    function DetailsReferenceController() {
         var ctl = this;
 
-        var ellipsis = $translate.instant('COMMON.ELLIPSIS');
-
         // Find the referenced object
         if (ctl.record) {
-            var references = _(ctl.record.data)
-                    .toArray()
-                    .flatten()
-                    .where({ _localId: ctl.data })
-                    .value();
+            // Scan through all items in the schema to find the matching reference
+            // by its _localId. The only piece of information we need about this
+            // reference is it's index, so it can be used in the auto-incrementing label.
+            var candidateIndices = _(ctl.record.data).toArray().filter(function(obj) {
+                return Array.isArray(obj) && obj.length > 0;
+            }).map(function(arr) {
+                return _.findIndex(arr, { _localId: ctl.data });
+            }).value();
 
-            // Construct a display label based on the properties of the referenced object
-            if (references.length) {
-                var reference = references[0];
-                var keys = _.without(_.keys(reference), '_localId', '$$hashKey');
-                var maxKeyLength = 12;
-                ctl.referenceDisplay = _.map(keys.slice(0, 3), function(key) {
-                    if (reference[key].length < maxKeyLength) {
-                        return reference[key];
-                    }
-                    // TODO: do we need to make changes for any displayed substrings
-                    // (and limitTo filters) in the app for right-to-left languages?
-                    return reference[key].substring(0, maxKeyLength) + ellipsis;
-                }).join(' ');
+            // There will only ever be one matching index. All other related info objects will
+            // have a -1 populated here, so we can just take the maxiumum of the array
+            var index = _.max(candidateIndices);
+            if (index > -1) {
+                ctl.referenceDisplay = ctl.property.propertyName + ' ' + (index + 1);
```

Comment:
Since we have the `if (index > -1) {` check, do we need a fallback branch in case no candidate matches?

Reply:
Talked with @kshepard and did some testing - it looks like situations where this falls through happens fairly frequently whenever the record is blank, so the default behavior leaving `referenceDisplay` unchanged seems perfectly fine.